src/keygen/Avvm.hs

```diff
@@ -130,3 +82,18 @@ genGenesis avvm genCerts holder = GenesisData
         let addr = makeRedeemAddress pk
             adaCoin = unsafeIntegerToCoin $ coinAmount coin
         return (addr, (adaCoin, distr adaCoin))
+
+getHolderId :: Maybe FilePath -> IO StakeholderId
+getHolderId holderPath = case holderPath of
+    Just fileName -> do
+        sk <- maybe
+            (fail "No secret key is found in file")
+            pure =<< view usPrimKey <$> readUserSecret fileName
```

Comment:
no, `fromMaybe a = maybe a identity`, but here `maybe a pure` is needed. Though I'll make it less obscure, got it.